registries/docker-daemon/configure.sh

```diff
@@ -3,15 +3,21 @@
 daemonConfig='/etc/docker/daemon.json'
 if test -f ${daemonConfig} && grep -q registry.kube-system.svc.cluster.local ${daemonConfig}; then
   echo "insecure registry previously configured"
-else
-  # Allow for insecure registries
-  sudo mkdir -p /etc/docker
-  echo '{ "insecure-registries": [ "registry.kube-system.svc.cluster.local:5000" ] }' | sudo tee ${daemonConfig} > /dev/null
-  sudo systemctl daemon-reload
-  sudo systemctl restart docker
+elif which systemctl > /dev/null; then
```

Comment:
seems like a fragile test. What's you're goal here?

Reply:
To not fail if there is no systemctl (e.g. in a container).

Reply:
As proposed, the docker daemon won't know that the registry is insecure and any attempt to push or pull images from the registry will fail.

That logic is being implemented somewhere outside of this script. There should be some other signal that the registry is already configured. Whether systemctl is available is not an indication that the registry was configured.

Reply:
It is, however, an indication that this script is going to fail. Everything works if the docker daemon is running on the host and you are in a container, but there is no need (or sense) then in trying to do anything with systemctl. We could use the /proc/1/cgroup test from the kind/start.sh script instead?

Reply:
See 6c45dea